Tp6m

klahnakoski opened the pull request:
This shows the pageload (and other suites of a single test) 

Requires https://github.com/mozilla-frontend-infra/firefox-health-dashboard/pull/291

klahnakoski commented:
Not including #291 and not including perf-goggles add

https://github.com/mozilla-frontend-infra/firefox-health-dashboard/pull/308/files/0ce79325e1544040596b5def8080c82e1790c15f..2a705889ade0f22f57037962c2882368a04eaeec?utf8=%E2%9C%93&diff=split&w=1

klahnakoski commented:
Adds a mobile page:

http://localhost:5000/quantum/tp6m?test=loadtime

Also allows the TP6 page to specify which (sub)test to display

http://localhost:5000/quantum/tp6?bits=64&test=loadtime

helfi92 commented on src/containers/PerfherderGraphContainer/index.jsx:
```diff
@@ -65,7 +65,7 @@ PerfherderGraphContainer.propTypes = {
         extraOptions: PropTypes.arrayOf(PropTypes.string),
         frameworkId: PropTypes.number.isRequired,
-        option: PropTypes.string.isRequired,
+        option: PropTypes.string,
```
For none required properties, define a default prop. Example:

```jsx
PerfherderGraphContainer.defaultProps = {
  option: null,
  // ...
}
```

klahnakoski replied:
This is more involved

klahnakoski replied:
https://github.com/mozilla-frontend-infra/firefox-health-dashboard/issues/312

helfi92 commented on src/quantum/index.jsx:
```diff
@@ -98,7 +98,7 @@ export default class QuantumIndex extends React.Component {
       {
         title: 'Page Load tests (TP6)',
-        more: `/quantum/tp6?bits=${bits}`,
+        more: `/quantum/tp6?bits=${bits}&test=pageload`,
```
https://deploy-preview-308--firefox-health-dashboard.netlify.com/quantum/tp6?bits=64&test=pageload shows:

![screen shot 2019-02-20 at 8 45 22 am](https://user-images.githubusercontent.com/3766511/53095910-eececd00-34eb-11e9-91a4-c5fd11b718c3.png)

klahnakoski replied:
thanks for catching that!

helfi92 commented on src/routes.jsx:
```diff
@@ -7,6 +7,7 @@ import JsTeam from './views/JsTeam';
 import NimbledroidGraphPage from './views/NimbledroidGraph';
 import Quantum from './quantum/index';
 import TP6 from './views/QuantumTP6/index';
+import TP6M from './views/QuantumTP6m/index';
```
No need to add `/index` at the end: `import TP6M from './views/QuantumTP6m';`. The index file is the default import.

helfi92 commented on src/utils/perfherder/chartJs/perfherderFormatter.js:
```diff
@@ -8,12 +8,16 @@ const dataToChartJSformat = data =>
-  const higherIsBetter = !series.meta.lower_is_better;
+  // TODO: map tests and suite scores to measurement units and
+  // add some label for scale
+  const isTest = !!series.meta.test;
```
`const isTest = Boolean(series.meta.test);`

klahnakoski commented:
@helfi92   one more review, since you caught mistakes in the last one, and I added links to the tp6m page

here are the changes since last review:

https://github.com/mozilla-frontend-infra/firefox-health-dashboard/pull/308/files/dc4988f4022b87eae7df1c4a968a9072fe78f9d2..4bdbce7ac5f670566e748666bdd1304f9658a286?utf8=%E2%9C%93&diff=split&w=1

helfi92 commented on src/routes.jsx:
```diff
@@ -6,8 +6,8 @@ import AndroidPage from './views/Android';
 import JsTeam from './views/JsTeam';
 import NimbledroidGraphPage from './views/NimbledroidGraph';
 import Quantum from './quantum/index';
```
`import Quantum from './quantum';`

klahnakoski replied:
https://deploy-preview-308--firefox-health-dashboard.netlify.com/quantum/tp6?bits=64&test=pageload  is expected to fail like this:  The URL has been fixed.

Concerning `import Quantum from './quantum/index';`  does your linter catch this?  Mine does not.

helfi92 replied:
I saw this when looking at the diff. Netlify previews ease the process of reviews such that I don't always need to pull the code down :)
